File: rust/src/chia_wallet/standart/puzzles/p2_conditions.rs
use crate::program_utils::program::Program;
use crate::program_utils::serialized_program::SerializedProgram;

pub fn p2_conditions_mod() -> SerializedProgram {
    SerializedProgram::from_hex("ff04ffff0101ff0280".to_string())
}

pub fn puzzleForConditions(conditions: Program) -> Program {
    let p2_program = p2_conditions_mod().to_program().unwrap();
    let mut args = Vec::new();
    args.push(conditions);
    let result = p2_program.run(Program::from(args));
    return result.program;
}

pub fn solution_for_conditions(conditions: Program) -> Program {
    return Program::from([puzzleForConditions(conditions), Program::from(0)].to_vec());
}

File: lib/classes/wallet_sdk/standart/puzzles/p2_conditions/p2_conditions.dart
// ignore_for_file: lines_longer_than_80_chars, non_constant_identifier_names

 
import '../../../../clvm/program.dart';

final P2_CONDITIONS_MOD = Program.deserializeHex(
  "ff04ffff0101ff0280",
);

Future<Program> puzzleForConditions(Program conditions) async{
  final result =  await P2_CONDITIONS_MOD.run([conditions]);
  return result.program;
}

Future<Program> solution_for_conditions(Program conditions) async{
  return await Program.list([await puzzleForConditions(conditions), Program.fromInt(0)]);
}
